File: Bai_71/main.cpp
#include<iostream>
#include<fstream>
#include<cmath>
using namespace std;

double last_digit_power(int m, int n){
    double last_digit = m % 10;
    if(last_digit == 0 || last_digit == 1 || last_digit == 5 || last_digit == 6) 
        return last_digit;
    double exp = n % 4;
    if(exp == 0) exp = 4;
    return static_cast<int>(pow(last_digit, exp)) % 10;
}

int main(){
    ifstream file("CHUSO.INP");
    ofstream file1("CHUSO.OUT");
    int m, n;
    file >> m >> n;
    file1 << last_digit_power(m, n);
    file.close();
    file1.close();
    return 0;
}

File: Bai_71/main.py
def last_digit_power(m, n):
    # Dùng tính chất của chữ số cuối cùng để giảm bớt số lần lặp
    last_digit = m % 10
    if last_digit == 0 or last_digit == 1 or last_digit == 5 or last_digit == 6:
        return last_digit
    
    # Sử dụng công thức Euler để tính toán chữ số cuối cùng
    exp = n % 4
    if exp == 0:
        exp = 4
    return pow(last_digit, exp) % 10

# Đọc dữ liệu từ file input
with open('CHUSO.INP', 'r') as f:
    m, n = map(int, f.readline().split())

# Tính toán chữ số cuối cùng của mũ n
result = last_digit_power(m, n)

# Ghi kết quả vào file output
with open('CHUSO.OUT', 'w') as f:
    f.write(str(result))